lib/liquid/tags/include.rb

```diff
@@ -44,7 +43,10 @@ def parse(tokens)
 
     def render(context)
       partial = load_cached_partial(context)
-      variable = context.evaluate(@variable_name)
+
+      template_name = context.evaluate(@template_name_expr)
+      variable = @variable_name_expr ? context.evaluate(@variable_name_expr) : context[template_name]
```

Comment:
Right... the only solution then is to create a VariableLookup manually or do what I had before..